### ISSUE TITLE
Linker search paths should be relative to `Cargo.toml`, not `$PWD`

### ISSUE THREAD

**Issue by DanielKeep** [rust-lang/cargo#1752]
Running `cargo build` causes Cargo to instruct `rustc` to search `.\rust\bin\$TARGET` and `bin\$TARGET` for libraries.  However, both of these are relative to the current working directory, *not* to the manifest.

Currently, this is biting me as it makes building from within my editor (Sublime Text 3) impossible without littering the source folders with duplicate `bin\$TARGET` folders.  I'd like to propose that the current behaviour is significantly less useful than having Cargo tell `rustc` to look in a predictable location for binary dependencies.

**Comment by alexcrichton** [comment]
This sounds like it may be coming from a build script instead of Cargo itself as these directories look somewhat unfamiliar to me, could you gist a full log of a failing build?

**Comment by DanielKeep** [comment]
@alexcrichton Here's [a short transcript](https://gist.github.com/DanielKeep/50d8ec3fa10200625c11) showing a build that fails when run from `src`, but *not* from along-side the `Cargo.toml`.  Just to be clear; `F:\Programming\Rust\repos\win32-bindgen` is the directory with the manifest in it.  There is *no* build script.  The dependency on `clang` is introduced by the source file `src/clang/ll.rs` which contains `#[link(name = "clang")]`.

**Comment by alexcrichton** [comment]
Can you paste the output of `cargo build -v` as well? I don't understand where the `-L` path for the clang.dll library is being added, and that may indicate whether this is a bug in Cargo or not.

**Comment by DanielKeep** [comment]
I updated the gist with [the output of `cargo build -v`](https://gist.github.com/DanielKeep/50d8ec3fa10200625c11#file-cargo-build-log) after running `cargo clean`.  You're right; those paths don't show up in the commands.

I was *convinced* that Cargo was responsible for these; that they were added as a work-around for Cargo unceremoniously deleting all the other local places it would search for libraries when building.

... it's actually `rustc` looking in the wrong spot, isn't it?  That's *really* bothersome because I thought this was a feature of Cargo, and I've been using it quite a lot.

I'll close this on the basis that it doesn't appear to be Cargo adding these paths at all.  *Damn.*
